packages/core/admin/admin/src/content-manager/components/DynamicTable/BulkActionsBar/index.js

```diff
@@ -240,16 +216,48 @@ const BulkActionsBar = ({
     }
   };
 
-  const handleConfirmUnpublishAll = async () => {
+  const handleBulkDelete = async () => {
     try {
       setIsConfirmButtonLoading(true);
-      await onConfirmUnpublishAll(selectedEntries);
+      await onConfirmDeleteAll(selectedEntries);
       clearSelectedEntries();
       setIsConfirmButtonLoading(false);
     } catch (err) {
       setIsConfirmButtonLoading(false);
     }
-    handleToggleShowUnpublishAllModal();
+    handleToggleShowDeleteAllModal();
+  };
+
+  const handleBulkPublish = () => {
+    setIsConfirmButtonLoading(true);
+
+    onConfirmPublishAll(selectedEntries, {
+      handleSuccess() {
+        clearSelectedEntries();
+        setIsConfirmButtonLoading(false);
+        handleToggleShowPublishAllModal();
+      },
+      handleError() {
+        setIsConfirmButtonLoading(false);
+        handleToggleShowPublishAllModal();
+      },
+    });
+  };
+
+  const handlBulkUnpublish = () => {
+    setIsConfirmButtonLoading(true);
+
+    onConfirmUnpublishAll(selectedEntries, {
+      handleSuccess() {
+        clearSelectedEntries();
+        setIsConfirmButtonLoading(false);
+        handleToggleShowUnpublishAllModal();
+      },
+      handleError() {
+        setIsConfirmButtonLoading(false);
+        handleToggleShowUnpublishAllModal();
+      },
```

Comment:
this feels very `then/catch` callback mayhem. What was the motivation behind not using something like `try/catch` or returning the data so you could handle it here? :)

Reply:
I didn't like try catch because we aren't actually doing anything with the error. 

I considered returning the data and handling it that way, and still could do that if you prefer, but we don't need the data so why return it. I was thinking this way the useMutation remains the single place handling the success and error states otherwise we are handling it in two places, right?

Reply:
Also if you did `try/catch/finally` you wouldn't need to do much for `error` to be honest.

Reply:
> I considered returning the data and handling it that way, and still could do that if you prefer, but we don't need the data so why return it. I was thinking this way the useMutation remains the single place handling the success and error states otherwise we are handling it in two places, right?

Sorry i didn't see this before i re-commented. I think the issue stems from how you've set it up. Right now you're using one mutation hook to do two things and if i'm honest i think it's more verbose this way. If you had separate mutations for `unpublish` and `publish` then the hooks could be setup:

```ts
const bulkPublishMutation = useMutation((data) => post(`/content-manager/collection-types/${contentType.uid}/actions/bulkPublish`, data), {
	onSuccess: () => {
		toggleNotification({
            type: 'success',
            message: { id: 'content-manager.success.record.publish', defaultMessage: 'Published' },
          });

          fetchData(`/content-manager/collection-types/${slug}${params}`);
	},
	onError: () => {
		 toggleNotification({
            type: 'warning',
            message: formatAPIError(error),
          });
	}
})
```

So the `onConfirmPublishAll` becomes:

```js
const handleConfirmPublishAllData = async (selectedEntries, { handleSuccess, handleError }) => {
    const validations = await validateEntriesToPublish(selectedEntries);

    if (Object.values(validations.errors).length) {
      toggleNotification({
        type: 'warning',
        title: {
          id: 'content-manager.listView.validation.errors.title',
          defaultMessage: 'Action required',
        },
        message: {
          id: 'content-manager.listView.validation.errors.message',
          defaultMessage:
            'Please make sure all fields are valid before publishing (required field, min/max character limit, etc.)',
        },
      });

      return;
    }

   return bulkPublishMutation.mutateAsync({ ids: selectedEntries });
  };
```

Which means your handler in `BulkActions` becomes:

```js
  const handleBulkPublish = async () => {
    setIsConfirmButtonLoading(true);
	
	try {
		await onConfirmPublishAll(selectedEntries)
		clearSelectedEntries();
	}
	catch(err){
		// the notification is handle in handleConfirmPublishAllData
	}
	finally {
		setIsConfirmButtonLoading(false);
        handleToggleShowPublishAllModal();
	}
  };
```

I think that's a bit clearer because we're not passing callbacks everywhere which IMO is how it's a bit tangled at the moment looking at the `helper-plugin` table 🤪

Reply:
Yeah I have a few ideas of how to handle this if we refactored e.g. we could use `useControllableState` for the selected items so if you need to know them and clear them (like you do) you control it from the parent, if you don't it's managed in the child.